src/main/java/com/dm/dmservicesandbox/controller/UserAuthController.java

```diff
@@ -0,0 +1,35 @@
+package com.dm.dmservicesandbox.controller;
+
+
+import com.dm.dmservicesandbox.domain.ServiceReponse;
+import com.dm.dmservicesandbox.domain.UserSignUpRequest;
+import com.dm.dmservicesandbox.service.UserAuthenticationService;
+import com.dm.dmservicesandbox.service.exception.UserAuthenticationException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import javax.validation.Valid;
+
+@Slf4j
+@RestController
+@RequestMapping("/dm/userAuth")
+public class UserAuthController {
+
+    @Autowired
+    UserAuthenticationService userAuthenticationService;
+
+    @PostMapping("/signup")
```

Comment:
prefer to have url name terminology same as method names, can we rename the method to be signupUser as well

Reply:
also if we want to be restful, we usually try to use the entity name for url with post. in this case if we wanted to be purist, we would name it POST /user (use /users). To be fair, I don't mind and we don't need to be 100% restful. I would tend to agree to have somewhat matching name (signup or register, but keep same in both)

Reply:
done renamed to registerUser